### ISSUE TITLE
Lua command to upgrade buildings

### ISSUE THREAD

**Issue by MarkMcWire** [widelands/widelands#4931]
Request to extend the Lua command set to include a command to upgrade buildings.

**Comment by MarkMcWire** [comment]
```
/* RST
   .. method:: enhance([keep_wares = false])

      Instantly enhance this building if there is an enhancement.

      :arg keep_wares: Optional: If :const:`false` (default) the wares in this buildings stock
         get destroyed. If :const:`true` the wares in this buildings stock will be preserved.
      :type keep_wares: :const:`bool`
*/
int LuaBuilding::enhance(lua_State* L) {
	Widelands::Building* bld = get(L, get_egbase(L));
	const Widelands::DescriptionIndex enhancement = bld->descr().enhancement();
	if (enhancement == Widelands::INVALID_INDEX) {
		return 0;
	}
	bld->get_owner()->enhance_building(bld, enhancement, lua_gettop(L) > 1 && luaL_checkboolean(L, 2));
	return 0;
}
```

**Comment by MarkMcWire** [comment]
```
class LuaBuilding : public LuaPlayerImmovable {
public:
	LUNA_CLASS_HEAD(LuaBuilding);

	LuaBuilding() {
	}
	explicit LuaBuilding(Widelands::Building& mo) : LuaPlayerImmovable(mo) {
	}
	explicit LuaBuilding(lua_State* L) : LuaPlayerImmovable(L) {
	}
	~LuaBuilding() override {
	}

	/*
	 * Properties
	 */
	int get_flag(lua_State* L);
	int get_destruction_blocked(lua_State* L);
	int set_destruction_blocked(lua_State* L);

	/*
	 * Lua Methods
	 */
	int dismantle(lua_State* L);
	int enhance(lua_State* L);

	/*
	 * C Methods
	 */
	CASTED_GET(Building)
};
```

**Comment by MarkMcWire** [comment]
```
/* RST
Building
--------

.. class:: Building

   Child of: :class:`PlayerImmovable`

   This represents a building owned by a player.

   More properties are available through this object's
   :class:`BuildingDescription`, which you can access via :any:`MapObject.descr`.
*/
const char LuaBuilding::className[] = "Building";
const MethodType<LuaBuilding> LuaBuilding::Methods[] = {
   METHOD(LuaBuilding, dismantle),
   METHOD(LuaBuilding, enhance),
   {nullptr, nullptr},
};
const PropertyType<LuaBuilding> LuaBuilding::Properties[] = {
   PROP_RO(LuaBuilding, flag),
   PROP_RW(LuaBuilding, destruction_blocked),
   {nullptr, nullptr, nullptr},
};
```

**Comment by MarkMcWire** [comment]
https://github.com/widelands/widelands/pull/4989/commits/1512f92b74879016ef629dbcfd5b75f12c53f988

**Comment by frankystone** [comment]
The issue will be closed if the #4989 got merged ;)
